core/primitives/src/transaction.rs

```diff
@@ -211,6 +211,27 @@ impl BorshDeserialize for Transaction {
     }
 }
 
+#[derive(Debug)]
+pub struct ValidatedTransaction(SignedTransaction);
+
+impl ValidatedTransaction {
+    pub fn new(tx: SignedTransaction) -> Result<Self, SignedTransaction> {
+        if tx.signature.verify(tx.get_hash().as_ref(), tx.transaction.public_key()) {
+            Ok(Self(tx))
+        } else {
+            Err(tx)
+        }
+    }
+
+    pub fn to_signed_transaction(&self) -> &SignedTransaction {
+        &self.0
+    }
+
+    pub fn into_signed_transaction(self) -> SignedTransaction {
+        self.0
+    }
```

Comment:
nit: Maybe just implement the Into trait for those two types?

Reply:
Had a similar discussion with Simonas about using `Deref`.  I have a bias towards accessors like these.  Using `Into` or `Deref` traits are fine as long as you have an IDE around but if you have to stare code on a random git branch on the terminal, then it can make things more challenging.  I added some more accessors and shortened the names further.